pkg/webhooks/walruscore/connector.go

```diff
@@ -2,42 +2,125 @@ package walruscore
 //
 // nolint: lll
+// +k8s:webhook-gen:mutating:group="walruscore.seal.io",version="v1",resource="connectors",scope="Namespaced"
+// +k8s:webhook-gen:mutating:operations=["CREATE","UPDATE"],failurePolicy="Fail",sideEffects="NoneOnDryRun",matchPolicy="Equivalent",timeoutSeconds=10
 // +k8s:webhook-gen:validating:group="walruscore.seal.io",version="v1",resource="connectors",scope="Namespaced"
 // +k8s:webhook-gen:validating:operations=["CREATE","UPDATE","DELETE"],failurePolicy="Fail",sideEffects="None",matchPolicy="Equivalent",timeoutSeconds=10
-type ConnectorWebhook struct{}
+type ConnectorWebhook struct {
+	client ctrlcli.Client
+}
 
 func (r *ConnectorWebhook) SetupWebhook(_ context.Context, opts webhook.SetupOptions) (runtime.Object, error) {
+	r.client = opts.Manager.GetClient()
+
 	return &walruscore.Connector{}, nil
 }
 
 var _ ctrlwebhook.CustomValidator = (*ConnectorWebhook)(nil)
 
 func (r *ConnectorWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) (ctrladmission.Warnings, error) {
-	// TODO: your logic here
+	return r.validateConnector(ctx, obj)
+}
+
+func (r *ConnectorWebhook) validateConnector(ctx context.Context, obj runtime.Object) (ctrladmission.Warnings, error) {
+	conn := obj.(*walruscore.Connector)
+	if stringx.StringWidth(conn.Name) > 30 {
+		return nil, field.TooLongMaxLength(field.NewPath("name"), stringx.StringWidth(conn.Name), 30)
+	}
+
+	err := resourcehandlers.IsConnected(ctx, conn, r.client)
+	if err != nil {
+		return nil, err
+	}
 
 	return nil, nil
 }
 
 func (r *ConnectorWebhook) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (ctrladmission.Warnings, error) {
-	// TODO: your logic here
-
-	return nil, nil
+	return r.validateConnector(ctx, newObj)
 }
 
 func (r *ConnectorWebhook) ValidateDelete(ctx context.Context, obj runtime.Object) (ctrladmission.Warnings, error) {
-	// TODO: your logic here
+	con := obj.(*walruscore.Connector)
+
+	list := new(walruscore.ConnectorBindingList)
+	err := r.client.List(ctx, list, &ctrlcli.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cb := range list.Items {
+		if cb.Spec.Connector.Name == con.Name && cb.Spec.Connector.Namespace == con.Namespace {
+			return nil, fmt.Errorf("connector is in use")
+		}
+	}
 
 	return nil, nil
 }
+
+func (r *ConnectorWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	conn := obj.(*walruscore.Connector)
+
+	data := map[string][]byte{}
+	for k, v := range conn.Spec.ConnectorConfig.Data {
+		data[k] = []byte(v.Value)
+	}
+
+	name := fmt.Sprintf("connector-config-%s-%s", conn.Namespace, conn.Name)
+	eSec := &core.Secret{
+		ObjectMeta: meta.ObjectMeta{
+			Namespace: conn.Namespace,
+			Name:      name,
+		},
+		Data: data,
+	}
+
+	conn.Spec.SecretName = name
+
+	hideSensitiveData(conn)
+	kubemeta.SanitizeLastAppliedAnnotation(conn)
+
+	_, err := kubeclientset.UpdateWithCtrlClient(ctx, r.client, eSec, kubeclientset.WithCreateIfNotExisted[*core.Secret]())
+
+	return err
```

Comment:
the secret name may exceed the limitation, use stringx.SumByFNV64a() to create a hash suffix.